### ISSUE TITLE
Can you provide a training set with annotated json and images

### ISSUE THREAD

**Issue by matesoul** [Beknaizer/OCT-Retinal-Layer-Segmenter#4]
Hello, author：
    I really want to learn from your project. I tried to annotate with the pictures and tools you provided to train the model, but the training results of my annotation were poor every time. Could you please provide a copy of the pictures and json files you annotated

**Comment by beasygo1ng** [comment]
Hello, I am very appreciated that you interested in my project. I can only share with simple example of dataset ( because have only this for now, i will try to annotate it soon ). Here the link to the google drive -> https://drive.google.com/file/d/1mZRxoPUn6eKbYrPi8jXua-nGhEftd1qa/view?usp=share_link/

P.S. Maybe i will make a video tutorial for clarification this part, because you're not first one who faced with this type of problem

**Comment by zk-416** [comment]
Hello author:
I used the link you provided, but the training results are still not very good. The first point is that my training loss and validation loss curves are very poor. And unable to obtain prediction images.
![prediction](https://user-images.githubusercontent.com/131513965/233767322-6231486d-53ee-4cfc-ac2b-04ead4ee75b1.png)
![training loss and validation loss](https://user-images.githubusercontent.com/131513965/233767327-8c322860-321c-42b8-9abc-31ce056d531b.png)
![prediction on tese image](https://user-images.githubusercontent.com/131513965/233767329-3c8bdc3b-ee6e-4244-9c51-38252becbfaf.png)

**Comment by zk-416** [comment]
Hello author:
I understand the error point of my problem is that the number of training times is too small. The number of training times in your source code is 20, but the number in the loss function curve is 200. I think you can comment. I can now run your code, thank you

**Comment by beasygo1ng** [comment]
Hi zk-416, 
Yes, **you pointed it out right.** Epochs in my source code is extremely less than in my original work, I will change it soon. But actually algorithm is work, if you are reviewed the code, in the output part we print transformed prediction 
`
prediction = (model.predict(test_img_input))`
`predicted_img = np.argmax(prediction, axis=3)[0,:,:]
`

But if we print out original _**prediction**_ instead of _**predicted_img**_, we can observe another result in below screenshot (amount of epochs is approximately same). We changed output argument from _**predicted_img**_  to  _**predicttion**_  in 20 line of code.  _**prediction**_  - it is what return out UNet model, numpy array  with shape [1,640,640,9] , where: 

- 1-> number of given input image(in our case single image)
- 640x640 -> size of our result image && designed input/output size of neural net (you can review it in first lines of code - SIZE_X,SIZE_Y. Actually it is up to you which size of result do you wanna set, but in this case you should also consider the amount of units in layers of UNet )
- 9 ->  amount of overall classes that we have ( it means that last index of array in print can be changed from 0 till 9, and it will print out corresponding layer)

![Screenshot from 2023-04-22 19-35-30](https://user-images.githubusercontent.com/70391405/233788225-b5f69e2c-5f36-419f-b55f-946dbcdb1970.png)

**Comment by beasygo1ng** [comment]
Also one moment that I wanna mention, size of annotated dataset that I shared upper is really small, you need larger set of images in order to make good model IRL. I recommend to you to do not split data to train, test, val(i know that it is not good practice) if you use my shared dataset. Use whole images to train, and for observe progress use not annotated images as test input (inference). You can get them anywhere ( i will attach the link to images that I used ). But you should understand that in this case, you can't take a actual metrics of performance of your model on new images.

OCTID datastet - https://dataverse.scholarsportal.info/dataset.xhtml?persistentId=doi:10.5683/SP/WLW4ZT

**Comment by zk-416** [comment]
Hello author:
Thank you very much for your reply. I think your project has been described in great detail. In the readme file, you shared the code for the OCT raw image dataset and added the annotated dataset to the answer. Users do not need to use makesense for annotation, nor do they need to run JSON_ mask_ reading.py。 You can run your code directly. I think you can make a note in the project to change the training frequency to 200 times.In this way, users can directly use and reproduce your project results. I think this may be more convenient.

**Comment by zk-416** [comment]
Hello author:
Thank you very much for your reply. I think your project has been described in great detail. In the readme file, you shared the code for the OCT raw image dataset and added the annotated dataset to the answer. Users do not need to use makesense for annotation, nor do they need to run JSON_ mask_ reading.py。 You can run your code directly. I think you can make a note in the project to change the training frequency to 200 times.In this way, users can directly use and reproduce your project results. I think this may be more convenient.

**Comment by beasygo1ng** [comment]
Yes [zk-416](https://github.com/zk-416),  I already commitedd new changes related to epochs and added link to the annotated dataset in readme file, thank you for contribution! And if you think that my project was useful for you, please rate it!
